cmd/xl-v1-object.go

```diff
@@ -198,8 +198,14 @@ func (xl xlObjects) GetObject(bucket, object string, startOffset int64, length i
 		return traceError(InvalidRange{startOffset, length, xlMeta.Stat.Size})
 	}
 
+	// Calculate endOffset according to length
```

Comment:
note that i/o does not hit because we wont enter the loop:
`for ; partIndex <= lastPartIndex; partIndex++ {`

also if we return in the object layer itself then we will miss out on checking the quorum error.